pkg/queue/manager.go

```diff
@@ -547,3 +594,105 @@ func (m *Manager) reportPendingWorkloads(cqName string, cq ClusterQueue) {
 	}
 	metrics.ReportPendingWorkloads(cqName, active, inadmissible)
 }
+
+func (m *Manager) Start(ctx context.Context) error {
+	defer m.queue.ShutDown()
+
+	queueVisibilityUpdateInterval := time.Duration(m.queueVisibilityUpdateInterval) * time.Second
+
+	for i := 0; i < snapshotWorkers; i++ {
+		go wait.UntilWithContext(ctx, m.takeSnapshot, queueVisibilityUpdateInterval)
+	}
+
+	go wait.UntilWithContext(ctx, m.enqueueSnapshotDelayed, queueVisibilityUpdateInterval)
+
+	<-ctx.Done()
```

Comment:
Is this needed? If so, please explain why.

Reply:
when we get context cancelation we will shutdown snapshot queue

Reply:
I see, this is analogous pattern to: https://github.com/kubernetes/kubernetes/blob/cd04f47e8b287068e8bf0bfae6596b2606d54929/pkg/controller/job/job_controller.go#L223.